modular_skyrat/modules/assault_operatives/code/vending_machine.dm

```diff
@@ -0,0 +1,77 @@
+//VENDING MACHINES>>>>>>>>>>>>>>>>>>>>>>>>>
+/obj/machinery/vending/assaultops_ammo
+	name = "\improper Syndicate Ammo Station"
+	desc = "An ammo vending machine which holds a variety of different ammo mags."
+	icon_state = "liberationstation"
+	vend_reply = "Item dispensed."
+	scan_id = FALSE
+	armor = list(MELEE = 100, BULLET = 100, LASER = 100, ENERGY = 100, BOMB = 0, BIO = 0, RAD = 0, FIRE = 100, ACID = 50)
+	resistance_flags = FIRE_PROOF
+	onstation = FALSE
+	light_mask = "liberation-light-mask"
+	default_price = 0
+	var/filled = FALSE
+
+/obj/machinery/vending/assaultops_ammo/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		fill_ammo(user)
+		ui = new(user, src, "Vending")
+		ui.open()
+
+
+/obj/machinery/vending/assaultops_ammo/proc/fill_ammo(mob/user)
+	if(last_shopper == user && filled)
+		return
+	else
+		filled = FALSE
+
+	if(!ishuman(user))
+		return FALSE
+
+	if(!user.mind.has_antag_datum(/datum/antagonist/assault_operative))
+		return FALSE
+
+	//Remove all current items from the vending machine
+	products.Cut()
+	product_records.Cut()
+
+	var/mob/living/carbon/human/human_user = user
+
+	//Find all the ammo we should display
+	for(var/i in human_user.contents)
```

Comment:
Please avoid using single-letter vars please.

Reply:
Tbh this can all be replaced with